content/en/user-guide/lambda-tools/debugging/index.md

```diff
@@ -418,6 +419,104 @@ $ awslocal lambda invoke --function-name func1 \
 {{% /tab %}}
 {{< /tabpane >}}
 
+## Lambda Debug Mode (Preview)
+
+Lambda Debug Mode is a preview feature in LocalStack designed to enhance your Lambda debugging workflows.
+This feature provides an optimized environment for debugging Lambda functions, ensuring that you have the
+necessary tools and flexibility to troubleshoot effectively.
+
+### Key Features
+* **Automatic Timeout Management**: Integrates with API Gateway to prevent Lambda function timeouts,
+giving developers ample time to connect remote debuggers and inspect the function's behavior.
+* **Multi-Function Debugging**: Supports debugging multiple Lambda functions concurrently.
+
+### Enabling Lambda Debug Mode
+
+To enable Lambda Debug Mode, set the `LAMBDA_DEBUG_MODE` environment variable as shown below:
+
+```bash
+$ LAMBDA_DEBUG_MODE=1 \
+  LAMBDA_DOCKER_FLAGS='-p 19891:19891' \
+  localstack start
+```
+
+When enabled, Lambda Debug Mode automatically adjusts timeouts to accommodate debugging needs:
+* **Lambda Container Startup Timeout**: Provides additional time for debugger connection during container creation.
+* **Lambda Execution Timeout**: Extends the execution window, allowing for in-depth remote debugging.
+* **API Gateway-Lambda Integration Timeout**: Increases timeout settings to avoid premature terminations.
+
+### Advanced Configuration
+
+For further customization, you can use a configuration file.
+Specify the path to this file with the `LAMBDA_DEBUG_MODE_CONFIG_PATH` environment variable.
+In most cases, manually setting `LAMBDA_DOCKER_FLAGS` is unnecessary when using this configuration.
+
+```bash
+$ LAMBDA_DEBUG_MODE=1 \
+  LAMBDA_DEBUG_MODE_CONFIG_PATH=debug_config.yaml \
```

Comment:
@MEPalma Don't we need to mount the debug config into the container as well?

Reply:
It's working for me without mounting 🥳 
Bundling debugpy and setting up everything is just so annoying 😿 Cross-building is important (if a sample does it barebone without Docker builds):

```
pip install \
--platform manylinux2014_x86_64 \
--target=src \
--implementation cp \
--python-version 3.10 \
--only-binary=:all: --upgrade \
debugpy
```

Without it, I got this error:

```
2024-08-28 21:06:34 0.00s - Debugger warning: It seems that frozen modules are being used, which may
2024-08-28 21:06:34 0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
2024-08-28 21:06:34 0.00s - to python to disable frozen modules.
2024-08-28 21:06:34 0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
2024-08-28 21:06:37 time="2024-08-28T19:06:37Z" level=error msg="Invoke failed" func=go.amzn.com/lambda/rapid.handleInvoke file="/home/runner/work/lambda-runtime-init/lambda-runtime-init/lambda/rapid/handlers.go:643" InvokeID=e45b4fb9-3082-46f6-b4a2-0339224064df error=errResetReceived
2024-08-28 21:06:37 time="2024-08-28T19:06:37Z" level=warning msg="Reset initiated: Timeout" func=go.amzn.com/lambda/rapid.handleReset file="/home/runner/work/lambda-runtime-init/lambda-runtime-init/lambda/rapid/handlers.go:710"
```

All good now 😄

Reply:
There are clearly many features and conveniences still to add to LDM. Regarding the bundling and configuring of remote debugging, this is something that aligns with the generalities of debugging lambda in LocalStack (or the previous approach), so as we already have documentations and example on this, I decided not to describe the setting up again for LDM.

Reply:
makes sense 👍  (just a random rant ;)